### PR TITLE
Handle create room action

### PR DESCRIPTION
**Pull request by Florian14**
#111 

### Content

- Create a room on the server and open it
- Show an error dialog with a retry button if the creation fails
- Wait for the room to be received from the sync before returning the result from the create room method

Note: Avatar upload will be handled separately

### CODE REVIEW

File: features/createroom/impl/src/main/kotlin/io/element/android/features/createroom/impl/CreateRoomFlowNode.kt [hunk 1]
```diff
@@ -74,7 +75,12 @@ class CreateRoomFlowNode @AssistedInject constructor(
                 createNode<CreateRoomRootNode>(context = buildContext, plugins = listOf(callback))
             }
             NavTarget.NewRoom -> {
-                createNode<ConfigureRoomFlowNode>(context = buildContext, plugins = emptyList())
+                val callback = object : ConfigureRoomNode.Callback {
+                    override fun onRoomCreated(roomId: RoomId) {
+                        plugins<CreateRoomEntryPoint.Callback>().forEach { it.onOpenRoom(roomId) }
```
Comment (bmarty) [inline features/createroom/impl/src/main/kotlin/io/element/android/features/createroom/impl/CreateRoomFlowNode.kt:80]: Maybe rename to `onRoomCreated` (it has more sense to me regarding the module contract)

```suggestion
                        plugins<CreateRoomEntryPoint.Callback>().forEach { it.onRoomCreated(roomId) }
```

Or maybe the module can return an existing room sometimes (ex: dm already exist)?

In this case, maybe rename to `onSuccess` or `onDone`.

File: features/createroom/impl/src/main/kotlin/io/element/android/features/createroom/impl/configureroom/ConfigureRoomPresenter.kt [hunk 2]
```diff
@@ -39,21 +52,47 @@ class ConfigureRoomPresenter @Inject constructor(
             }
         }
 
+        val localCoroutineScope = rememberCoroutineScope()
+        val createRoomAction: MutableState<Async<RoomId>> = remember { mutableStateOf(Async.Uninitialized) }
+
+        fun createRoom(config: CreateRoomConfig) {
+            createRoomAction.value = Async.Uninitialized
```
Comment (bmarty) [inline features/createroom/impl/src/main/kotlin/io/element/android/features/createroom/impl/configureroom/ConfigureRoomPresenter.kt:59]: Nit: setting to `Uninitialized` could be done in the fun `createRoom()`

**Review by bmarty — APPROVED** [review]
LGTM, thanks!

File: features/createroom/impl/src/main/kotlin/io/element/android/features/createroom/impl/configureroom/ConfigureRoomPresenter.kt (hunk 2 above)
Comment (Florian14) [reply features/createroom/impl/src/main/kotlin/io/element/android/features/createroom/impl/configureroom/ConfigureRoomPresenter.kt]: https://github.com/vector-im/element-x-android/pull/324/commits/f566a543e4913b84e6ea596e685ee1dafe2eeeea

File: features/createroom/impl/src/main/kotlin/io/element/android/features/createroom/impl/CreateRoomFlowNode.kt (hunk 1 above)
Comment (Florian14) [reply features/createroom/impl/src/main/kotlin/io/element/android/features/createroom/impl/CreateRoomFlowNode.kt]: https://github.com/vector-im/element-x-android/pull/324/commits/2403775857c8b791a64a4e107a25c5bbd9cd6abc

**Pull request merged**